ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -881,17 +887,38 @@ def runTest(self):
                 self.watching = False
 
             if self.reboot_type == 'warm-reboot':
+                # Pause data/control plane watchers while the test is running
+                self.log("Pause watching DUT")
+                self.watch_dataplane = False
+                self.watch_controlplane = False
+                self.watcher_dp_is_stopped.wait(timeout = 5)  # Wait for the dataplane Watcher stopped.
+                self.watcher_cp_is_stopped.wait(timeout = 5)  # Wait for the dataplane Watcher stopped.
```

Comment:
what if 5 minutes is not enough?

Reply:
5 is 5 seconds. Nothing happens when timeout is reached, the test goes further.
It just gives a time (max 5 seconds) for the reachability_watcher() to stop current _while_ iteration (i.e. to stop printing "Send %5d Received %5d ping DUT" and "Send %5d Received %5d t1->servers").
Without the wait timeout, if reachability_watcher() for some reason failed to stop, wait() will wait forever and hang up.

Reply:
What is the reason for stopping control plane watcher here and resume it at line 905?

Reply:
But what if 5 seconds is not enough to stop? Might it broke the script?

Reply:
> What is the reason for stopping control plane watcher here and resume it at line 905?

I stop reachability_watcher() while send_and_sniff() is running, otherwise they influence each other.
After send_and_sniff() is finished, I resume reachability_watcher() for Control Plane only.

Reply:
> But what if 5 seconds is not enough to stop? Might it broke the script?

Its the same when _time.sleep(5)_ is not enough.
_wait(timeout)_ faits for the Event to happen. If the Event didn't happen during _timeout_, script goes further.
It a graceful replace of explicit _time.sleep(5)_.

Reply:
But sending data plane ping takes 2-3 minutes. By the time when it is done. The control plane is long back. Then we didn't really get the benefit of knowing when did the control plane come back. In that case, then maybe this change is not needed?

Reply:
The log (in the description) shows that the Control Plane is back ~2 minutes later after the test finished send_and_snif.
At least this watcher confirms that Control Plane is back and the DUT is finally up (otherwise fails with the relevant message).